common/utils/rawUnitConversion.ts

```diff
@@ -0,0 +1,24 @@
+import { fromTokenBase, getDecimalFromEtherUnit, TokenValue, Wei, UnitKey } from 'libs/units';
+
+interface Args {
+  value?: TokenValue | Wei | null;
+  symbol?: string;
+}
+
+interface EthProps extends Args {
+  unit: UnitKey;
+}
+interface TokenProps extends Args {
+  decimal: number;
+}
+
+// export const rawUnitConversion = (args: Args & (EthProps | TokenProps)) => {
+export const rawUnitConversion = (args: Args & (EthProps | TokenProps)) => {
+  const { value } = args;
+  const isEthereumUnit = !!(args as EthProps).unit;
+  return value
+    ? isEthereumUnit
+      ? fromTokenBase(value, getDecimalFromEtherUnit((args as EthProps).unit))
```

Comment:
Should use type guards instead of type coercion as seen in `UnitDisplay` L39 (`IsEthereumUnit`)